### PR TITLE
chore: add workflow to create testing-prelease testing builds

### LINKED ISSUE #8957: Provide a testing builds of the Podman Desktop to be used across our CI
### Is your enhancement related to a problem? Please describe

Right now we are mostly running e2e tests against podman desktop built in development mode. We would like to expand the e2e tests to be run also against production builds of Podman Desktop. We need to enable electron inspect flag during the build to produce such builds. these builds then can be use during e2e test in a way, where we will simply download the bits and we will not be building whole project from scratch everywhere.

### Describe the solution you'd like

Similar to standard nightly builds of Podman Desktop we are releasing to https://github.com/containers/podman-desktop-prereleases repository, we could create repository, ie. `podman-desktop-testing-prereleases` where we would publish podman desktop production builds with `ELECTRON_ENABLE_INSPECT` option enabled, so that we can use them in the e2e testing workflow/pipelines.

Create a repository, add/adjust step to builds also nightly with this env.var set and publish it in given repo from within particular nightly workflow.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Workflow create a testing builds of podman desktop (with enabled `ELECTRON_ENABLE_INSPECT` flag) for mac, linux and windows - it produces executable files (tar.gz, exe and dmg). It pushes the artifacts as a prerelease on testing-prereleases repository with given tag based on latest main commit pushed to the main branch in podman-desktop repo. Runs daily at 1 am and can be triggered on demand. Builds will be used in automated testing especially in extensions and when running E2E tests on nightly base in various CI.

There is an idea to improve the workflow iteratively with other improvements depending on need.
### What issues does this PR fix or reference?
Fixes #8957 

### How to test this PR?
Hard to simulate the run and push  of the artifacts somewhere from main repo. Poc: https://github.com/odockal/testing-prereleases/blob/main/.github/workflows/testing-build.yaml 

- [ ] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Comment by odockal** [comment]
@benoitf I think we also need a token to be able to push to testing-prereleases repo. Can you please take a look?

File: .github/workflows/daily-testing-build.yaml [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  workflow_dispatch:
+  schedule:
+    - cron: '0 1 * * *'
+
+env:
+  GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+  DEBUG: electron-builder
+
+jobs:
+
```
Comment (gastoner) [inline .github/workflows/daily-testing-build.yaml]: Extra new line?

**Review by gastoner — APPROVED** [review]
Codewise LGTM

**Comment by odockal** [comment]
> @benoitf I think we also need a token to be able to push to testing-prereleases repo. Can you please take a look?

I think I am not able to use personal token to access necessary resources on testing-prereleases in podman-desktop org. I would need someone with maintainer/owner access to provide a token so we can publish releases there. Can you please take a look?

**Comment by benoitf** [comment]
@odockal as you used the PODMAN_DESKTOP_BOT_TOKEN it's already set

**Comment by odockal** [comment]
@benoitf great then, can I merge?

**Comment by benoitf** [comment]
yes

**Pull request merged**